src/pages/spawn/spawnVolume/spawnVolumeTable/spawnVolumeTableActions/migrateBtn/migrateVolumeModal/reducer.ts

```diff
@@ -0,0 +1,37 @@
+import { FormState } from "components/Spawn/spawnHostModal";
+
+export enum Page {
+  First,
+  Second,
+}
+
+interface State {
+  page: Page;
+  form: FormState;
+}
+
+export const initialState = { page: Page.First, form: {} };
+
+export function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case "goToNextPage":
+      return {
+        ...state,
+        page: state.page === Page.First ? Page.Second : Page.First,
+      };
+    case "resetPage":
+      return { ...state, page: Page.First };
+    case "setForm":
+      return { ...state, form: action.payload };
+    case "resetForm":
+      return { ...state, form: {} };
+    default:
+      throw new Error();
```

Comment:
Lets add an error message here just so it shows up in bugsnag.
```suggestion
      throw new Error(`Unknown reducer action ${action}`);
```